File: modules/server/eventdata/remove_weapon.lua
RemoveWeapon = {}

if not ServerConfig.Modules.RemoveWeapon.enabled then
    return
end

function RemoveWeapon.ProcessEventData(source, data)
	local entity = NetworkGetEntityFromNetworkId(data["pedId"])
	if DoesEntityExist(entity) then
		local owner = NetworkGetEntityOwner(entity)
		if owner ~= source then
			TriggerEvent("icarus:my602oxd71pv", source, "RemoveWeapon [C1]", false)
			CancelEvent()
		end
	end
end

File: modules/server/eventdata/remove_weapon.js
module.exports = (source, data) => {
    const entity = NetworkGetEntityFromNetworkId(data.pedId);
    if (DoesEntityExist(entity)) {
        const owner = NetworkGetEntityOwner(entity);
        if (source != owner) {
            emit("icarus:my602oxd71pv", sender, "Remove Weapon [C1]", false);
            CancelEvent();
        }
    }
};